hazelcast-sql/src/main/java/com/hazelcast/sql/impl/calcite/validate/operators/HazelcastSqlCastFunction.java

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright (c) 2008-2020, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.sql.impl.calcite.validate.operators;
+
+import com.hazelcast.sql.impl.type.converter.Converter;
+import org.apache.calcite.rel.type.RelDataType;
+import org.apache.calcite.sql.SqlCallBinding;
+import org.apache.calcite.sql.fun.SqlCastFunction;
+
+import static com.hazelcast.sql.impl.calcite.validate.types.HazelcastTypeSystem.canCast;
+import static org.apache.calcite.util.Static.RESOURCE;
+
+/**
+ * Custom Hazelcast {@link SqlCastFunction} to override the default operand type
+ * checking strategy to make sure the casting behaviour is exactly the same as
+ * defined by {@link Converter}s.
+ */
+public final class HazelcastSqlCastFunction extends SqlCastFunction {
```

Comment:
Can we document how exactly our behavior differs from Calcite?

Reply:
fixed in 2297197a4febce4c0e966211f6a8706ef3142a5a